### PR TITLE
Change ChangeLease return type

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Comment by ahsonkhan** [comment]
Please hold off on merging any other PRs for the next couple of hours.

**Comment by katmsft** [comment]
> Please hold off on merging any other PRs for the next couple of hours.

I assume we are gtg right?

**Comment by ahsonkhan** [comment]
> I assume we are gtg right?

Yes.

File: sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_lease_client.hpp [hunk 1]
```diff
@@ -57,29 +57,30 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
...
 
     /**
      * @brief Changes the lease of an active lease.
      *
      * @param leaseId ID of the previously-acquired lease.
      * @param proposedLeaseId Proposed lease ID, in a GUID string format.
      * @param options Optional parameters to execute this function.
-     * @return A ChangeShareLeaseResult describing the lease.
+     * @return A ChangeShareLeaseResult describing the updated lease.
+     * @remarks The current ShareLeaseClient becomes invalid if this operation succeeds.
```
Comment (katmsft) [inline sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_lease_client.hpp:79]: Not only if the operation succeeds, but also when the operation fails for some particular failures, including the response is not received for a successful result or if the server times out but the change is done before the timeout.
How about `The current ShareLeaseClient becomes invalid or undefined state after this function is called.`

**Review by katmsft — APPROVED** [review]

File: sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_lease_client.hpp (hunk 1 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_lease_client.hpp:79]: I believe it's a general rule that the result is undetermined (meaning we cannot assume the result being successful or failed) if errors like network failure happens

**Pull request merged**